### DISCUSSION TITLE
Middleware does not work correctly in NextJS 13

### DISCUSSION THREAD

**Discussion by darushHamidi** [vercel/next.js#46439] [Help] [answered]
I have the following **middleware**. middleware says if the URL starts with` "/admin" `run `console.log('We are in the admin route...').`
In development, It works great. When I build the project(`npm run build`) and then start it (`npm run start`)
the middleware does not work correctly. When I navigate to` http://localhost:3000`
we will see console.log('We are in the admin route...') in the terminal.
Why? It has to log (`console.log('We are in the admin route...')`) when the URL starts with `"/admin"`

```
import { NextRequest, NextResponse } from 'next/server';

const isAdminRoute = (pathname: string) => {
  return pathname.startsWith('/admin');
}

export async function middleware(req: NextRequest) {
  const { pathname } = req.nextUrl;

  if (isAdminRoute(pathname)) {
    console.log(pathname)
    console.log('We are in the admin route...')
    return NextResponse.redirect(new URL('/login', req.url));
  }

  return NextResponse.next();
}

```

[Stackblitz Is Here.](https://stackblitz.com/edit/nextjs-skdfyj)
If you see the project and build it you will see the problem.

**Thanks in advance**

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I think the problem is happening because the NavBar component uses `next/link`, which does automatic prefetching.

Prefetching is only enabled in production builds.

You can disable it, but it will still happen on hover. https://nextjs.org/docs/api-reference/next/link

> prefetch - Prefetch the page in the background. Defaults to true. Any <Link /> that is in the viewport (initially or through scroll) will be preloaded. Prefetch can be disabled by passing prefetch={false}. When prefetch is set to false, prefetching will still occur on hover. Pages using [Static Generation](https://nextjs.org/docs/basic-features/data-fetching/get-static-props) will preload JSON files with the data for faster page transitions. Prefetching is only enabled in production.
